### PR TITLE
[Backport] MAGETWO-61422 Respect Category Top Navigation Max Depth setting

### PR DESCRIPTION
**Pull request by torhoehn**
### Original Pull Request 
 https://github.com/magento/magento2/pull/12640
### Description
The setting `Stores > Configuration > Catalog > Catalog > Category Top Navigation > Maximal Depth` was actually never being used to limit the top menu maximum depth.

Now the setting will be used as a condition for the category collection when it is fetched.

### Fixed Issues
1. magento/magento2#7543: Category Top Navigation / Maximal Depth configuration not working

### Manual testing scenarios
1. Change the setting in the backend
2. Refresh the invalidated cache
3. See that the setting actually is being used

### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All new or changed code is covered with unit/integration tests (if applicable)
 - [x] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Review by orlangur — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @orlangur, thank you for the review.
[ENGCOM-4716](https://jira.corp.magento.com/browse/ENGCOM-4716) has been created to process this Pull Request

**Comment by VasylShvorak** [comment]
:heavy_check_mark: QA passed

File: app/code/Magento/Catalog/Model/ResourceModel/Category/Collection.php [hunk 1]
```diff
@@ -402,6 +460,23 @@ public function addRootLevelFilter()
         return $this;
     }
 
+    /**
+     * Add navigation max depth filter
+     *
+     * @return $this
+     */
+    public function addNavigationMaxDepthFilter()
```
Comment (sivaschenko) [inline app/code/Magento/Catalog/Model/ResourceModel/Category/Collection.php:468]: Adding public methods to API classes in patch releases is prrrohibited by our backward compatibility policy

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Thanks for the backport @torhoehn ! However, the pull request contains several backward incompatible and not suitable for patch upgrade changes (introduced public functions and constructor parameters)

Considering the above,  wouldn't backport this change to 2.2-develop branch. Please let me know what you think

cc @orlangur

**Comment by sidolov** [comment]
@torhoehn , I am closing this PR now due to inactivity.
Please reopen and update if you wish to continue.
Thank you for the collaboration!

**Pull request closed without merging**